system/jlib/jtrace.cpp

```diff
@@ -69,3 +100,688 @@ const char* LogTrace::queryLocalId() const
+        //}
+    }
+
+public:
+    CServerSpan(const char * spanName, const char * tracerName_, StringArray & httpHeaders)
+    : CSpan(spanName, SpanType::ServerSpan, tracerName_)
+    {
+        setSpanContext(httpHeaders);
+        init();
+    }
+
+    CServerSpan(const char * spanName, const char * tracerName_, const IProperties * httpHeaders)
+    : CSpan(spanName, SpanType::ServerSpan, tracerName_)
+    {
+        setSpanContext(httpHeaders);
+        init();
+    }
+};
+
+class CTraceManager : implements ITraceManager, public CInterface
+//class CTraceManager : CInterfaceOf<ITraceManager>
+{
+private:
+    bool enabled = true;
+    StringAttr moduleName;
+
+    void initTracer(IPropertyTree * traceConfig)
+    {
+        try
+        {
+#ifdef TRACECONFIGDEBUG
+            if (!traceConfig || !traceConfig->hasProp("tracing"))
+            {
+                const char * simulatedGlobalYaml = R"!!(global:
+    tracing:
+        enable: true
+        exporterx:
+          type: OTLP
+          endpoint: "localhost:4317"
+          useSslCredentials: true
+          sslCredentialsCACcert: "ssl-certificate"
+        processor:
+          batchSpan: true
+          simpleSpan: false
+    )!!";
+                Owned<IPropertyTree> testTree = createPTreeFromYAMLString(simulatedGlobalYaml, ipt_none, ptr_ignoreWhiteSpace, nullptr);
+                traceConfig = testTree->getPropTree("global/tracing");
+            }
+
+            StringBuffer xml;
+            toXML(traceConfig, xml);
+            DBGLOG("traceConfig tree: %s", xml.str());
+#endif
+
+            if (traceConfig && traceConfig->getPropBool("@enable", false))
+            {
+                DBGLOG("OpenTel tracing enabled");
+                using namespace opentelemetry::trace;
+                //std::unique_ptr<opentelemetry::sdk::trace::SpanExporter> exporter = opentelemetry::exporter::trace::OStreamSpanExporterFactory::Create();
+                //std::unique_ptr<opentelemetry::sdk::trace::SpanExporter> exporter = opentelemetry::sdk::trace::export::NoopSpanExporter;
+
+                //Currently using InMemorySpanExporter as default, until a noop exporter is available
+                std::shared_ptr<opentelemetry::exporter::memory::InMemorySpanData> data;
+                std::unique_ptr<opentelemetry::sdk::trace::SpanExporter> exporter = opentelemetry::exporter::memory::InMemorySpanExporterFactory::Create(data);
+
+                Owned<IPropertyTree> exportConfig = traceConfig->getPropTree("exporter");
+#ifdef TRACECONFIGDEBUG
+                StringBuffer xml;
+                toXML(exportConfig, xml);
+                DBGLOG("exportConfig tree: %s", xml.str());
+#endif
+                if (exportConfig)
+                {
+                    StringBuffer exportType;
+                    exportConfig->getProp("@type", exportType);
+                    DBGLOG("Exporter type: %s", exportType.str());
+
+                    if (!exportType.isEmpty())
```

Comment:
Is this all connected to publishing to an external service?  It might be useful to split this into separate functions and/or add comments to explain what it is setting up/why it is needed.

Reply:
> Is this all connected to publishing to an external service? It might be useful to split this into separate functions and/or add comments to explain what it is setting up/why it is needed.

Yes, in part. The concept of exporting span data (to memory, std streams, log, file, prometheous, etc) is built into the Otel tracer provider and almost immediately SCapps wants to control the exporting of trace info to a collector of their choosing. I believe they intend to export over OTLP... I'll comment as much as I can inline and remove this logic from this PR via ifdef with the intent to enable once the helm schema is defined.